### DISCUSSION TITLE
Linux下打包出来的产物含有~demos/:id/index.html，提交到git后在Windows系统无法拉取代码

### DISCUSSION THREAD

**Discussion by alienzhangyw** [umijs/dumi#2009] [Q&A] [answered]
首先我项目中并没有使用到demo功能，只用markdown写文档。
之前都是在自己本地Windows打包，文件夹里不会生成~demos目录和文件。

现在迁移到Github Actions自动打包，把产物自动提交到服务器用的仓库。我发现打包出来的目录有个~demos/:id/index.html，因为路径存在冒号，导致在Windows下无法同步代码了。

尝试改git设置`git config core.protectNTFS false`也不起作用。

这个是dumi的feature还是bug？

**Comment by PeachScript** [comment, marked as answer]
是 feature，部分托管平台是可以支持动态路由的，但因为 windows 的文件系统不支持 `:` 所以 dumi 在 *niux 系统下会生成 `:id` 文件夹，你如果最终产物是要给 windows 服务器托管，那么 GitHub CI 也使用 windows 的机器跑即可

**Reply by alienzhangyw** [reply]
服务器也是Linux，只不过我需要在Windows下拉取代码修改一些配置

**Reply by actake** [reply]
补充：目前项目里是没有动态路由的，linux上还会打包出:id目录；是否能够优化下有动态路由才打包该目录？
